solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -70,6 +73,11 @@
 
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+  /**
+   * A counter to ensure that no RID is equal, even if they fall in the same millisecond
+   */
+  private static final AtomicLong ridCounter = new AtomicLong();
```

Comment:
LongAdder is probably a better choice

Reply:
👍 I'll look into it.  Tbh I moved this from DebugComponent and didn't consider alternatives much.

Reply:
I don't think LongAdder is appropriate here - it doesn't have an incrementAndGet or anything similar that would ensure threads get unique counter values.

The "atomicity" that it offers, afaict, is that calls to `add(long)` can't be clobbered across threads and will eventually be incorporated into the sum().  Nothing guarantees that two threads can't get the same `sum()` value at a given time, without additional synchronization.  I found this [SO post](https://stackoverflow.com/questions/45269587/does-java-longadders-increment-sum-prevent-getting-the-same-value-twice) helpful.

Now that I look at the synchronization generally though it reinforces my impression that I need to benchmark this before I commit, to ensure the lock contention isn't noticeable.  It's a shame LongAdder doesn't look like a good fit.

Reply:
It's really not a big deal and I'd be shocked if it noticeably impacted performance.

Reply:
I agree, but I wanted an excuse to try out the perf [framework](https://github.com/thesearchstack/solr-bench) that Ishan has been teasing off and on for awhile, so I tested it anyways.  With 30 threads submitting queries in parallel it added ~1.5ms to the p95 times.

Sounds pretty insignificant to me.  I added a flag query param to disable it in any case.